chef/cookbooks/updater/recipes/default.rb

```diff
@@ -45,6 +48,67 @@
       ignore_failure true
     end
 
+    ruby_block "check for updates" do
+      block do
+        case node[:updater][:zypper][:method]
+        when "patch"
+          command = "list-patches"
+        when "update"
+          command = "list-updates"
+        when "dist-upgrade"
+          command = "list-updates"
+        end
+
+        node.run_state["needs_update"] = `zypper -q #{command}|wc -l`.chomp.to_i > 0
+
+        command += '|egrep -q "corosync|pacemaker"'
+        system("zypper #{command}")
+        # exit 0: found, 1 not found
+        node.run_state["found_ha_packages"] = $?.exitstatus ? true : false
+      end
+      not_if { search(:node, "roles:pacemaker-cluster-member").empty? }
+    end
+
+    ["corosync", "pacemaker"].each do |s|
+      service s do
+        action :stop
+        only_if { node.run_state["found_ha_packages"] }
+        not_if { node[:pacemaker][:is_remote] }
```

Comment:
I think you should make sure that `node[:pacemaker]` exists; If I'm not missing sometthing, this could could be executed at non-pacemaker nodes as well